### PR TITLE
Update default razor project cache file name

### PR DESCRIPTION
**Pull request by phil-allen-msft**
Based on https://github.com/dotnet/razor/pull/9270

### CODE REVIEW

File: src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/RazorWorkspaceListenerInitializer.cs [hunk 1]
```diff
@@ -14,7 +14,7 @@ internal sealed class RazorWorkspaceListenerInitializer
 {
     // This should be moved to the Razor side once things are announced, so defaults are all in one
     // place, in case things ever need to change
-    private const string _projectRazorJsonFileName = "project.razor.vscode.json";
+    private const string _projectRazorJsonFileName = "project.razor.vscode.bin";
```
Comment (dibarbet) [inline src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/RazorWorkspaceListenerInitializer.cs:17]: Would be good to move this to the razor side entirely (and have Roslyn consume it via an input param or option) as mentioned in the comment.  But no need to do now if this needs to go in urgently.

**Review by dibarbet — APPROVED** [review]
does this need to go into the pending vscode release as well?

File: src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/RazorWorkspaceListenerInitializer.cs (hunk 1 above)
Comment (davidwengier) [inline src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/RazorWorkspaceListenerInitializer.cs:17]: Good idea to update this, but its only a fallback and probably will never be used. The real value to use gets passed in on the command line here: https://github.com/dotnet/vscode-csharp/blob/main/src/razor/src/razorLanguageServerClient.ts#L245

I took the liberty: https://github.com/dotnet/vscode-csharp/pull/6447

Comment (davidwengier) [reply src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/RazorWorkspaceListenerInitializer.cs:17]: I was wrong, this is used, but only for writing the file. Reading the file needs the other PR I opened. Sorry, holiday brain.

**Pull request merged**